catalog/task/push-snapshot/0.9/tests/mocks.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env sh
+set -eux
+
+# mocks to be injected into task step scripts
+
+function cosign() {
+  echo Mock cosign called with: $*
+  echo $* >> $(workspaces.data.path)/mock_cosign.txt
```

Comment:
Your setup task declares a `config` workspace, not `data`. But somehow it still works? Any idea how? Either I'm missing something, or it sounds like a bug that somehow it exposes the `data` workspace even if it's only defined for another task.

In any case, the workspace names should match. You could use different workspace names for setup and check-results, but it's probably best to stick with one name.

Reply:
Because `tests-workspace` is provided in both cases, just using different names per task. But, I didn't mean to be inconsistent. I will update it

Reply:
I know it points to the same data, but that doesn't explain why it works. In the context of the first task, I would expect `$(workspaces.data.path)` to be undefined - so basically either empty, or kept as that string. Because there is no such workspace in this task. It's as if Tekton was thinking: "Hmm, we don't have this workspace in this task, but I can see a workspace of this name is defined in another task in this pipeline, so I guess I'll use that one."

Reply:
Huh? My setup task had this
```
  workspaces:
    - name: tests-workspace
  tasks:
    - name: setup
      workspaces:
        - name: data
          workspace: tests-workspace
      taskSpec:
        workspaces:
          - name: data
        steps:
          - name: setup-values
            image: quay.io/hacbs-release/release-utils:85ab98a7ec63c3d8d9ec3c4982ff0e581bcb3c83
            script: |
              #!/usr/bin/env sh
              set -eux

              cat > $(workspaces.data.path)/mapped_snapshot.json << EOF
```
and my check-result had
```
    - name: check-result
      workspaces:
        - name: data
          workspace: tests-workspace
      taskSpec:
        workspaces: 
          - name: data
```
I am using tests-workspace for both. I just gave it different names (accidentally)
```

Reply:
Ah, sorry, my bad. I mixed it up. Somehow I was thinking that mocks.sh and the setup task have to match, but `mocks.sh` is injected into the actual task (`run-task`) and that task's workspace was called `data` from the start. So all was good. (But yeah, it's still nice to use one name everywhere.)